backend/models/user.model.ts

```diff
@@ -57,18 +57,27 @@ const UserSchema: Schema = new Schema({
 });
 
 /* eslint-disable func-names */
-UserSchema.pre("findOneAndDelete", async function () {
-  const doc = await this.findOne(this.getQuery()).clone();
-  if (doc.role !== "Teacher") return;
+UserSchema.pre("findOneAndDelete", async function (next) {
+  try {
+    const doc = await this.findOne(this.getQuery()).clone();
+    if (doc && doc.role === "Teacher") {
+      // Delete teacher reference from associated school
+      await MgSchool.findOneAndUpdate(
+        { teachers: doc.id },
+        { $pull: { teachers: doc.id } },
+      );
+
+      // Delete classes associated with teacher
+      await MgClass.deleteMany({ teacher: doc.id });
+
+      // Delete test sessions associated with teacher
+      await MgTestSession.deleteMany({ teacher: doc.id });
+    }
+  } catch (error) {
+    return next(error as CallbackError | undefined);
+  }
 
-  /* eslint-disable no-underscore-dangle */
-  await MgSchool.findOneAndUpdate(
-    { teachers: doc._id },
-    { $pull: { teachers: doc._id } },
-    { new: true },
-  );
-  await MgTestSession.deleteMany({ teacher: doc._id });
-  await MgClass.deleteMany({ teacher: doc._id });
+  return next();
```

Comment:
also here ^. have we lumped in all the delete dangling reference tickets into this PR? want to clarify the scope of these changes

Reply:
same as above